src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -414,4 +515,42 @@ void LRUFileCachePriority::shuffle(const CachePriorityGuard::Lock &)
         queue.splice(queue.end(), queue, it);
 }
 
+std::string LRUFileCachePriority::getStateInfoForLog(const CachePriorityGuard::Lock & lock) const
+{
+    return fmt::format("size: {}/{}, elements: {}/{} (description: {})",
+                       getSize(lock), max_size, getElementsCount(lock), max_elements, description);
+}
+
+void LRUFileCachePriority::holdImpl(
+    size_t size,
+    size_t elements,
+    const CachePriorityGuard::Lock & lock)
+{
+    chassert(size || elements);
+
+    if (!canFit(size, elements, lock))
+    {
+        throw Exception(ErrorCodes::LOGICAL_ERROR,
+                        "Cannot take space {} in size and {} in elements. "
+                        "Current state {}/{} in size, {}/{} in elements",
+                        size, elements, state->current_size, max_size,
+                        state->current_elements_num, max_elements);
+    }
+
+    state->current_size += size;
+    state->current_elements_num += elements;
+
+    // LOG_TEST(log, "Hold {} by size and {} by elements", size, elements);
```

Comment:
There are a few commented out log messages in this PR, they all were added in this PR and are very useful for debugging, but I commented them out because they are too verbose even for `test` log level, also as they slow down execution (noticeable in stress tests), they might prevent some bugs from being caught. Still I comment them out instead of just deleting, because I want to be able to add them back quickly if I need to debug something again.